src/maps.rs

```diff
@@ -141,10 +150,40 @@ where
 }
 
 /// Parse the maps file for the process with the given PID.
-pub(crate) fn parse(pid: Pid) -> Result<Vec<MapsEntry>, Error> {
+pub(crate) fn parse<F>(pid: Pid, filter: F) -> Result<Vec<MapsEntry>, Error>
+where
+    F: FnMut(&MapsEntry) -> bool,
+{
     let path = format!("/proc/{pid}/maps");
     let file = File::open(path)?;
-    parse_file(file, pid)
+    parse_file(file, pid, filter)
+}
+
+/// A helper function checking whether a `MapsEntry` has relevant to
+/// symbolization efforts. If that is not the case, it may be possible to ignore
+/// it altogether.
+pub(crate) fn is_symbolization_relevant(entry: &MapsEntry) -> bool {
+    // Only entries with actual paths are of relevance.
+    if entry.path.as_path().components().next() != Some(Component::RootDir) {
```

Comment:
this seems too eager, e.g., '[vdso]' is relevant

Reply:
It's just a code move, we are preserving semantics here.

Reply:
yeah, that's fine, just pointing out such corner cases that spring to my mind when I look at this code :)